hwbench/environment/vendors/vendor.py

```diff
@@ -31,6 +32,22 @@ def __del__(self):
         if self.logged:
             self.redfish_obj.logout()
 
+    def add_monitoring_value(
+        self,
+        monitoring_struct: dict[str, dict[str, MonitorMetric]],
+        context: Any,
+        metric: MonitorMetric,
+        name: str,
+        value: float,
+    ) -> dict[str, dict[str, MonitorMetric]]:
+        """This function add a new <value> in the monitoring data structure."""
+        if str(context) not in monitoring_struct:
+            monitoring_struct[str(context)] = {}
```

Comment:
Optional: I am wondering if making monitoring_struct a DefaultDict can help so that it's always at least an empty directory.